### PR TITLE
Update contour.py and widget.py

### PR DESCRIPTION
**Pull request by ZhaoZhonglun1991**
## PR Summary
Make MultiCursor could draw different cursors between the current axes which mouse stay and other axes.
## PR Checklist

- [x] Has Pytest style unit tests
- [x] Code is PEP 8 compliant
- [x] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/contour.py [hunk 1]
```diff
@@ -1406,7 +1406,12 @@ def find_nearest_contour(self, x, y, indices=None, pixel=True):
                     ymin = xc[1]
 
         return (conmin, segmin, imin, xmin, ymin, dmin)
-
+  def remove(self):
```
Comment (anntzer) [inline lib/matplotlib/contour.py]: why is this needed?

Comment (tacaswell) [reply lib/matplotlib/contour.py]: Because it does not have this method and it is rather annoying to sort out how to remove contours objects.

**Comment by tacaswell** [comment]
Thanks for your work on this @ZhaoZhonglun1991 !

We follow (and enforce via CI testing) a sub-set of PEP8 on the code base.  If you look at the bottom of the travis logs it will tell you exactly what is wrong.

Could you provide a bit more explanation as to _why_ you are proposing these changes?

File: lib/matplotlib/contour.py (hunk 1 above)
Comment (ZhaoZhonglun1991) [reply lib/matplotlib/contour.py]: To avoid reploting Basemap when we want to remove contours objects and plot a new one.

**Comment by jklymak** [comment]
? It looks like maybe you have combined two PRs here?

**Comment by ZhaoZhonglun1991** [comment]
Because this is the first time I tried to pull requests , I wasn't familiar with it.

**Comment by jklymak** [comment]
Sure - keep each PR on a separate branch and then checkout each branch as you work on it, and then push each branch.  The PR will automatically update.

**Comment by ZhaoZhonglun1991** [comment]
Thank you. Should I resend these two PRs?

**Comment by jklymak** [comment]
If you can clean things up in your local branches, you can then just push them to the GitHub and they will be cleaned up here as well.  You may have got yourself in such a state that its hard to do that, so not the end of the world to close and re-open new PRs.  What I usually do if I've really made a mess, is copy my messed up branch into a new branch (locally), delete the old branch (locally), and then make a fresh "old branch" where I fix things.  A local backup sometimes helps :wink:  

*Never* never never do work on your local master branch.  In fact you should probably delete it.  

Now that you've seen how things can go awry, have a look at https://matplotlib.org/devdocs/devel/gitwash/index.html

**Comment by ZhaoZhonglun1991** [comment]
I really appreciate the help that you have given me.

**Comment by xiazhujie** [comment]
I think you copy my idea about multi cursor via QQ
 group and commit after my PR(#10273)  is not very good,bro.

**Comment by jklymak** [comment]
This seems abandoned and/or maybe duplicated.  Closing.  Feel free to ask for a re-open....
